drivers/regulator/regulator_max20335.c

```diff
@@ -0,0 +1,405 @@
+
+enum max20335_pmic_sources {
+	MAX20335_PMIC_SOURCE_BUCK1,
+	MAX20335_PMIC_SOURCE_BUCK2,
+	MAX20335_PMIC_SOURCE_LDO1,
+	MAX20335_PMIC_SOURCE_LDO2,
+	MAX20335_PMIC_SOURCE_LDO3,
+};
+
+struct regulator_max20335_desc {
+	uint8_t vsel_reg;
+	uint8_t enable_mask;
+	uint8_t enable_val;
+	uint8_t cfg_reg;
+	const struct linear_range *uv_range;
+	const struct linear_range *ua_range;
+};
+
+struct regulator_max20335_config {
+	struct regulator_common_config common;
+	struct i2c_dt_spec bus;
+	const struct regulator_max20335_desc *desc;
+	uint8_t source;
+};
+
+struct regulator_max20335_data {
+	struct regulator_common_data common;
+};
+
+static const struct linear_range buck1_range = LINEAR_RANGE_INIT(700000, 25000U, 0x0U, 0x3FU);
+static const struct linear_range buck2_range = LINEAR_RANGE_INIT(700000, 50000U, 0x0U, 0x3FU);
+static const struct linear_range buck12_current_limit_range =
+	LINEAR_RANGE_INIT(50000, 25000U, 0x02U, 0x0FU);
+static const struct linear_range ldo1_range = LINEAR_RANGE_INIT(800000, 100000U, 0x0U, 0x1CU);
+static const struct linear_range ldo23_range = LINEAR_RANGE_INIT(900000, 100000U, 0x0U, 0x1FU);
+
+static const struct regulator_max20335_desc buck1_desc = {
+	.vsel_reg = MAX20335_BUCK1_VSET,
+	.enable_mask = MAX20335_BUCK_EN_MASK,
+	.enable_val = MAX20335_BUCK_EN,
+	.cfg_reg = MAX20335_BUCK1_CFG,
+	.uv_range = &buck1_range,
+	.ua_range = &buck12_current_limit_range,
+};
+
+static const struct regulator_max20335_desc buck2_desc = {
+	.vsel_reg = MAX20335_BUCK2_VSET,
+	.enable_mask = MAX20335_BUCK_EN_MASK,
+	.enable_val = MAX20335_BUCK_EN,
+	.cfg_reg = MAX20335_BUCK2_CFG,
+	.uv_range = &buck2_range,
+	.ua_range = &buck12_current_limit_range,
+};
+
+static const struct regulator_max20335_desc ldo1_desc = {
+	.vsel_reg = MAX20335_LDO1_VSET,
+	.enable_mask = MAX20335_LDO_EN_MASK,
+	.enable_val = MAX20335_LDO_EN,
+	.cfg_reg = MAX20335_LDO1_CFG,
+	.uv_range = &ldo1_range,
+};
+
+static const struct regulator_max20335_desc ldo2_desc = {
+	.vsel_reg = MAX20335_LDO2_VSET,
+	.enable_mask = MAX20335_LDO_EN_MASK,
+	.enable_val = MAX20335_LDO_EN,
+	.cfg_reg = MAX20335_LDO2_CFG,
+	.uv_range = &ldo23_range,
+};
+
+static const struct regulator_max20335_desc ldo3_desc = {
+	.vsel_reg = MAX20335_LDO3_VSET,
+	.enable_mask = MAX20335_LDO_EN_MASK,
+	.enable_val = MAX20335_LDO_EN,
+	.cfg_reg = MAX20335_LDO3_CFG,
+	.uv_range = &ldo23_range,
+};
+
+static int regulator_max20335_set_enable(const struct device *dev, bool enable)
+{
+	const struct regulator_max20335_config *config = dev->config;
+
+	switch (config->source) {
+	case MAX20335_PMIC_SOURCE_BUCK1:
+		__fallthrough;
+	case MAX20335_PMIC_SOURCE_BUCK2:
+		__fallthrough;
+	case MAX20335_PMIC_SOURCE_LDO1:
+		__fallthrough;
+	case MAX20335_PMIC_SOURCE_LDO2:
+		__fallthrough;
+	case MAX20335_PMIC_SOURCE_LDO3:
+		return i2c_reg_update_byte_dt(&config->bus,
+				      config->desc->cfg_reg,
+				      config->desc->enable_mask,
+				      enable ? config->desc->enable_val : 0);
```

Comment:
all switch/cases look redundant now (source is initialized by the driver itself.. and seems to handle all used values)

Reply:
Remove redundant parts.